src/components/Toggle/Toggle.module.css

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+.toggle {
+  display: grid;
+  width: 44px;
+  height: 24px;
+}
+
+.toggle > * {
+  grid-area: 1/1;
+  width: 44px;
+  height: 24px;
+  box-sizing: border-box;
+}
+
+.toggle > input[type="checkbox"] {
+  opacity: 0;
+  margin: 0;
+  cursor: pointer;
+}
+
+.toggle > input[type="checkbox"][disabled] {
+  cursor: not-allowed;
+}
+
+.toggle-ui {
+  pointer-events: none;
+  width: 44px;
+  aspect-ratio: 1 / 1;
+  border-radius: 9999px;
+  border: 1px solid var(--cpd-color-border-interactive-primary);
+  position: relative;
+  padding: 1px;
+  transition: all 0.2s ease-in-out;
+}
+
+:checked + .toggle-ui {
+  background: var(--cpd-color-bg-action-primary-rest);
+}
+
+[type="checkbox"]:hover + .toggle-ui {
+  border-color: var(--cpd-color-border-interactive-hovered);
+}
+
+:checked:hover + .toggle-ui {
+  background: var(--cpd-color-bg-action-primary-hovered);
+}
+
+.toggle-ui::after {
+  content: "";
+  display: block;
+  width: 20px;
+  aspect-ratio: 1 / 1;
+  border-radius: 50%;
+  background: var(--cpd-color-bg-action-primary-rest);
+  transform: translateX(0);
+  transition: all 0.2s ease-in-out;
+}
+
+[type="checkbox"]:focus + .toggle-ui {
+  outline: 2px solid var(--cpd-color-border-focused);
+  outline-offset: 1px;
+}
+
+:checked + .toggle-ui::after {
+  transform: translateX(100%);
+  background: var(--cpd-color-icon-on-solid-primary);
+}
+
+[type="checkbox"][disabled] + .toggle-ui {
+  border-color: var(--cpd-color-border-disabled);
+  background: var(--cpd-color-bg-canvas-disabled);
+}
+
+[type="checkbox"][disabled] + .toggle-ui::after {
+  background: var(--cpd-color-bg-action-primary-disabled);
+}
+
+[type="checkbox"][disabled]:checked + .toggle-ui {
+  background: var(--cpd-color-bg-action-primary-disabled);
+}
```

Comment:
From @janogarcia 

> The checked + disabled state is not rendered correctly, it results in a plain gray surface. The token mapping for [type="checkbox"][disabled] + .toggle-ui::after is wrong. Also, the border should be transparent, as noted above.

Reply:
The background should not be transparent, according to the design it is the same color as the background. I've made the adjustment! 

Thank you for the heads up on the visual glitch!